datafusion-examples/examples/flight/flight_server.rs

```diff
@@ -98,7 +99,7 @@ impl FlightService for FlightServiceImpl {
                 let df = ctx.sql(sql).await.map_err(to_tonic_err)?;
 
                 // execute the query
-                let schema = df.schema().clone().into();
+                let schema: SchemaRef = df.schema().into();
```

Comment:
Likewise here and below, I would find it easier 

```suggestion
                let schema: SchemaRef = Arc::clone(df.schema.as_ref());
```

Reply:
This PR adds `impl From<&DFSchema> for SchemaRef`. I can retract it, no problem.
However,   it will make it easier to reach out to pre-existing `impl From<&DFSchema> for Schema` and wrap in `Arc::new` (instead of `Arc::clone`).
The question is not how to write correct cheap arc cloning code. The question is how to write cheap code while not thinking about this specifically.